plugins/inputs/mysql/mysql.go

```diff
@@ -920,74 +861,210 @@ func (m *Mysql) GatherUserStatisticsStatuses(db *sql.DB, serv string, acc telegr
 		return err
 	}
 	defer rows.Close()
-	var (
-		user                   string
-		total_connections      int64
-		concurrent_connections int64
-		connected_time         int64
-		busy_time              int64
-		cpu_time               int64
-		bytes_received         int64
-		bytes_sent             int64
-		binlog_bytes_written   int64
-		rows_fetched           int64
-		rows_updated           int64
-		table_rows_read        int64
-		select_commands        int64
-		update_commands        int64
-		other_commands         int64
-		commit_transactions    int64
-		rollback_transactions  int64
-		denied_connections     int64
-		lost_connections       int64
-		access_denied          int64
-		empty_queries          int64
-		total_ssl_connections  int64
-		count                  uint32
-	)
+
+	cols, err := convertColumns(rows.Columns())
+	if err != nil {
+		return err
+	}
+
+	read, err := getColSlice(len(cols))
+	if err != nil {
+		return err
+	}
+
+	if len(read) != len(cols) {
```

Comment:
I would probably remove this check, since in theory it is impossible unless there is a programming error and it should. be picked up by the rows.Scan(

Reply:
:+1: